lib/iris/tests/unit/analysis/test_PERCENTILE.py

```diff
@@ -208,12 +208,25 @@ def test_masked(self):
         shape = (2, 11)
         data = ma.arange(np.prod(shape)).reshape(shape)
         data[0, ::2] = ma.masked
-        emsg = "Cannot use fast np.percentile method with masked array."
+        emsg = (
+            "Cannot use fast np.percentile method with masked array unless "
+            "mdtol is 0."
+        )
         with self.assertRaisesRegex(TypeError, emsg):
             PERCENTILE.aggregate(
                 data, axis=0, percent=50, fast_percentile_method=True
             )
 
+    def test_masked_mdtol_0(self):
+        shape = (2, 11)
+        axis = 0
+        percent = 50
+        data = ma.arange(np.prod(shape)).reshape(shape)
+        data[0, ::2] = ma.masked
+        expected = np.arange(shape[-1]) + 5.5
+        expected[0] = ma.masked
+        self.check_percentile_calc(data, axis, percent, expected, mdtol=0)
```

Comment:
I think we need to do something cleverer here as `check_percentile_calc` uses `assertArrayEqual`, which ignores masks.  There is also [assertMaskedArrayEqual](https://github.com/SciTools/iris/blob/caae45075795c1e13fbb0f04488c3b99f15b60e8/lib/iris/tests/__init__.py#L686) which could be what we want.  I made a mini test script to check

```python
import iris.tests as tests

import numpy as np
import numpy.ma as ma

class TestDummy(tests.IrisTest):
    def setUp(self):
        self.arr1 = np.arange(3)
        self.arr2 = ma.arange(3)
        self.arr2[0] = ma.masked
        
    def test_equal(self):
        self.assertArrayEqual(self.arr1, self.arr2)
        
    def test_masked_equal(self):
        self.assertMaskedArrayEqual(self.arr1, self.arr2)
```

`pytest` gave me
```
FAILED array_equal_check.py::TestDummy::test_masked_equal - AssertionError: 
========================= 1 failed, 1 passed in 2.20s ==========================
```

Reply:
Done (though I now have 4 test failures)